jtec-core/src/main/java/edu/tum/sse/jtec/reporting/LCOVReportParser.java

```diff
@@ -0,0 +1,147 @@
+                    continue;
+                final String sourceFilePath = sourceFile.toString();
+
+                final String sourceCode = IOUtils.readFromFile(sourceFile);
+                final CompilationUnit compilationUnit = StaticJavaParser.parse(sourceCode);
+                final ClassOrInterfaceDeclaration classDeclaration = compilationUnit.getClassByName(className).orElse(null);
+                if (classDeclaration == null)
+                    continue;
+
+                LCOVSection lcovSection = lcovReport.getLcovSection(sourceFilePath);
+                if (lcovSection == null)
+                    continue;
+                final List<MethodDeclaration> methodDeclarations = (methodName != null) ? classDeclaration.getMethodsByName(methodName) : classDeclaration.getMethods();
+                for (MethodDeclaration methodDeclaration : methodDeclarations) {
+                    final Range range = methodDeclaration.getRange().orElse(null);
+                    if (range == null)
+                        continue;
+                    lcovSection.addMethodHit(1, methodDeclaration.getNameAsString());
+                    for (int line = range.begin.line; line <= range.end.line; line++)
+                        lcovSection.addLineHit(line, 1);
+                }
+            }
+        }
+        return lcovReport;
+    }
+
+    /**
+     * For each method in each source file, create an LCOV section with hit count 0 for all lines.
+     * @return The list of initializd LCOV sections
+     */
+    private List<LCOVSection> getInitializedLcovSections() throws IOException {
+        List<LCOVSection> lcovSections = new ArrayList<>();
+        for (File file : sourceFiles) {
+            final LCOVSection lcovSection = new LCOVSection("", file.getAbsolutePath());
+            final String sourceCode = IOUtils.readFromFile(file.toPath());
+            final CompilationUnit compilationUnit = StaticJavaParser.parse(sourceCode);
+            List<MethodDeclaration> methodDeclarations = compilationUnit.findAll(MethodDeclaration.class);
+            for (MethodDeclaration methodDeclaration : methodDeclarations) {
+                final Range range = methodDeclaration.getRange().orElse(null);
+                if (range == null)
+                    continue;
+                lcovSection.addMethod(range.begin.line, methodDeclaration.getNameAsString());
+                for (int line = range.begin.line; line <= range.end.line; line++)
+                    lcovSection.addLineHit(line, 0);
+            }
+            lcovSections.add(lcovSection);
+        }
+        return lcovSections;
+    }
+
+    /**
+     * Get the Java source file corresponding to a class
+     * @param className The fully qualified class name
+     * @return The source file if it exists, else null
+     */
+    private Path getJavaSourceFile(String className) {
+        final String relativeFilePath = className.replace('.','/') + ".java";
```

Comment:
we unfortunately cannot assume that the file path to the .java file is the same as the .class file.
say, for instance the following `src/main/java/foo/App.java` is compiled, it results in a class `bar.App`:
```java
package bar;

class App {}
```

Reply:
We need to search all files in the project's source directory and parse all .java files to find the .java file containing the correct full-qualified class name...